### PR TITLE
Add table oci_core_cluster_network closes #449

### LINKED ISSUE #449: Add table oci_core_cluster_network
**References**
https://pkg.go.dev/github.com/oracle/oci-go-sdk/v44/core#ComputeManagementClient.GetClusterNetwork

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from oci_core_cluster_network
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------
| id                                                                                                | display_name                  | lifecycle_state | time_created              | time_updated              | instance_pools                                                
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------
| ocid1.clusternetwork.oc1.ap-mumbai-1.amaaaaaa6igdexaay55ef4foqlogh6fa52zd556rbyfznpoknbfp6enblbvq | cluster-network-20230111-2100 | PROVISIONING    | 2023-01-11T21:00:39+05:30 | 2023-01-11T21:02:12+05:30 | [{"availabilityDomains":["TvRS:AP-MUMBAI-1-AD-1"],"compartment
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------

Time: 6.7s. Rows fetched: 1. Hydrate calls: 1.
> .cache off
> .cache clear
> select * from oci_core_cluster_network where display_name = 'cluster-network-20230111-2100'
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------
| id                                                                                                | display_name                  | lifecycle_state | time_created              | time_updated              | instance_pools                                                
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------
| ocid1.clusternetwork.oc1.ap-mumbai-1.amaaaaaa6igdexaay55ef4foqlogh6fa52zd556rbyfznpoknbfp6enblbvq | cluster-network-20230111-2100 | PROVISIONING    | 2023-01-11T21:00:39+05:30 | 2023-01-11T21:02:12+05:30 | [{"availabilityDomains":["TvRS:AP-MUMBAI-1-AD-1"],"compartment
+---------------------------------------------------------------------------------------------------+-------------------------------+-----------------+---------------------------+---------------------------+---------------------------------------------------------------


```
</details>

### CODE REVIEW

File: oci/table_oci_core_cluster_network.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+)
+
+//// TABLE DEFINITION
+
+func tableCoreClusterNetwork(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_cluster_network",
+		Description: "OCI Core Cluster Network",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
```
Comment (karanpopat) [inline oci/table_oci_core_cluster_network.go:22]: Error in the GET call
```sql
> select * from oci_core_cluster_network where id = 'sad'

Error: Error returned by ComputeManagement Service. Http Status Code: 400. Error Code: InvalidParameter. Opc request id: 8ecd1440dde62b774e8f9751376e4dc8/DD6BBB87D31F75AA14BFA3AB3517CB6D/FBDD5B0C3AA878FFEA97D7B4E2C82067. Message: path param clusterNetworkId Invalid OCID.  Found: sad
Operation Name: GetClusterNetwork
Timestamp: 2023-01-16 07:26:44 +0000 GMT
Client Version: Oracle-GoSDK/65.28.0
Request Endpoint: GET https://iaas.ap-mumbai-1.oraclecloud.com/20160918/clusterNetworks/sad
Troubleshooting Tips: See https://docs.oracle.com/iaas/Content/API/References/apierrors.htm#apierrors_400__400_invalidparameter for more information about resolving this error.
Also see https://docs.oracle.com/iaas/api/#/en/iaas/20160918/ClusterNetwork/GetClusterNetwork for details on this operation's requirements.
To get more info on the failing request, you can set OCI_GO_SDK_DEBUG env var to info or higher level to log the request/response details.
If you are unable to resolve this ComputeManagement issue, please contact Oracle support and provide them this full error message. (SQLSTATE HV000)
```

File: docs/tables/oci_core_cluster_network.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  id,
+  time_created,
+  lifecycle_state as state
+from
+  oci_core_cluster_network
+where
+  lifecycle_state = 'AVAIALABLE';
+```
+
+### Lis cluster networks created in last 30 days
```
Comment (karanpopat) [inline docs/tables/oci_core_cluster_network.md:47]:
```suggestion
### List cluster networks created in the last 30 days
```

**Review by karanpopat — APPROVED** [review]
LGTM

File: oci/table_oci_core_cluster_network.go [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+					Name:    "display_name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "lifecycle_state",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildCompartementZonalList,
```
Comment (bigdatasourav) [inline oci/table_oci_core_cluster_network.go:42]: Can we use BuildCompartementList here?

**Pull request merged**